### PR TITLE
[onert/odc] Introduce minmax embedder

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit intrdocues minmax data embedder for circle model file into on-device quantizer. 
It uses raw file format to load minmax data, ant it will be saved by runtime.

ONE-DCO-1.0-Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Draft: #12903
Related issue: https://github.com/Samsung/ONE/issues/12904

### CODE REVIEW

File: runtime/onert/odc/Embedder.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+{
+namespace odc
+{
+struct EmbedderOptions
+{
+  float min_percentile = 0.0f; // dummy initial value to make SE tool happy
+  float max_percentile = 0.0f; // dummy initial value To make SE tool happy
+};
+
+class Embedder
```
Comment (hseok-oh) [inline runtime/onert/odc/Embedder.h:34]: Almost same with `compiler/minmax-embedder/include/minmax-embedder/Embedder.h`, except using `luci::Module`.

File: runtime/onert/odc/MinMaxReader.cc [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      {
+        // Read minmax data
+        readMMFile(minmax, sizeof(float), 2, file, "Cannot read minmax data from file");
+        mmv.min_vector.emplace_back(minmax[0]);
+        mmv.max_vector.emplace_back(minmax[1]);
+
+        // Skip remain input minmax data
+        uint32_t remain_elem = num_input - i - 1;
+        std::fseek(file, data_size * remain_elem, SEEK_CUR);
+        break;
```
Comment (zetwhite) [inline runtime/onert/odc/MinMaxReader.cc:154]: Just for curiosity, 

is it necessary to move SEEK_CUR to the end?
Just closing the file without moving SEEK_CUR to the end is a bad idea?

Comment (zetwhite) [reply runtime/onert/odc/MinMaxReader.cc:154]: aha, I got it. 
It is because there are `runs` times iterations.

**Review by zetwhite — APPROVED** [review]
LGTM :+1:

**Review by jyoungyun — APPROVED** [review]
LGTM

File: runtime/onert/odc/Embedder.cc [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+  if (percentile == 0.0)
+    return copy.front();
+
+  if (percentile == 100.0)
+    return copy.back();
+
+  int index = static_cast<int>(std::floor((copy.size() - 1) * percentile / 100.0));
+
+  float percent_i = static_cast<float>(index) / static_cast<float>(copy.size() - 1);
```
Comment (ragmani) [inline runtime/onert/odc/Embedder.cc:63]: Is there any case where `copy.size() - 1` is `0`?
Even so, how about adding an assertion that checks `copy.size() - 1` is not `0`?

**Review by ragmani — APPROVED** [review]
LGTM

File: runtime/onert/odc/Embedder.cc (hunk 3 above)
Comment (hseok-oh) [reply runtime/onert/odc/Embedder.cc:63]: Good point. I'll update this later.

Comment (hseok-oh) [reply runtime/onert/odc/Embedder.cc:63]: If `copy.size()` is 1, then `vector.size()` is also 1, so this function return `vector[0]` and does not reach here.

**Pull request merged**